contract/bridge/contracts/VaultReward.sol

```diff
@@ -0,0 +1,186 @@
+    }
+  }
+
+  function updateVaultAccClaimableRewards(address _vaultId) external onlyOneBtc {
+    if (block.timestamp <= lockedVaults[_vaultId].lockExpireAt) {
+      // get vault
+      LockedVault storage vault = lockedVaults[_vaultId];
+
+      // store the old accClaimableRewards
+      uint256 oldAccClaimableRewards = vault.accClaimableRewards;
+
+      // update the vault info
+      (uint256 claimableRewards, uint256 rewardClaimAt) = getClaimableRewards(_vaultId);
+      vault.accClaimableRewards = claimableRewards;
+      vault.rewardClaimAt = rewardClaimAt;
+
+      emit UpdateVaultAccClaimableRewards(_vaultId, oldAccClaimableRewards, vault.accClaimableRewards, rewardClaimAt);
+    }
+  }
+
+  function claimRewards(address _vaultId) external {
+    require(_vaultId == msg.sender, "Invalid vaultId");
+
+    // get reward debt
+    (uint256 claimableRewards, uint256 rewardClaimAt) = getClaimableRewards(_vaultId);
+    
+    // update the vault info
+    LockedVault storage vault = lockedVaults[_vaultId];
+    vault.accClaimableRewards = 0;
+    vault.rewardClaimAt = rewardClaimAt;
+
+    // transfer rewards
+    IVaultReserve(vaultReserve).withdrawReward(msg.sender, claimableRewards);
+
+    emit ClaimRewards(_vaultId, claimableRewards, rewardClaimAt);
+  }
+
+  function getClaimableRewards(address _vaultId) public view vaultExist(_vaultId) returns (uint256 claimableRewards, uint256 rewardClaimAt) {
+    // get vault collateral
+    (,, uint256 collateral,,,,,, uint256 liquidatedCollateral) = IVaultRegistry(oneBtc).getVault(_vaultId);
+    uint256 vaultUsedCollateral = collateral.sub(liquidatedCollateral);
+
+    // get vault
+    LockedVault memory vault = lockedVaults[_vaultId];
+
+    // get APR based on the lock period
+    uint256 lockPeriod = vault.lockPeriod;
+    uint256 vaultAPR;
+    if (lockPeriod == 3) {
+      vaultAPR = 5;
+    } else if (lockPeriod >= 6) {
+      vaultAPR = 10;
+    } else if (lockPeriod >= 12) {
+      vaultAPR = 15;
+    }
+
+    // get the elapsed time
+    uint256 secPerDay = 60 * 60 * 24;
+    uint256 elapsedSecs;
+    if (block.timestamp <= vault.lockExpireAt) {
+      rewardClaimAt = block.timestamp.sub(mod(block.timestamp.sub(vault.rewardClaimAt), secPerDay.mul(14)));
+      elapsedSecs = rewardClaimAt.sub(vault.rewardClaimAt);
+    } else {
+      rewardClaimAt = vault.lockExpireAt;
+      elapsedSecs = rewardClaimAt.sub(vault.rewardClaimAt);
+    }
+
+    // calculate the remaining rewards since the last claim time
+    uint256 claimableUnit = elapsedSecs.div(secPerDay).div(14); // claim every 14 days
+    if (claimableUnit == 0) {
+      claimableRewards = vault.accClaimableRewards;
+    } else {
+      claimableRewards = vault.accClaimableRewards.add(vaultUsedCollateral.mul(claimableUnit).mul(14).mul(vaultAPR).div(36500));
+    }
+  }
+
+  function mod(uint256 a, uint256 b) private pure returns (uint256 remainder) {
```

Comment:
why not use `mod` function from SafeMathUpgradeable directly?

Reply:
Fixed.

Reply:
Fixed